nautobot/extras/tests/test_views.py

```diff
@@ -909,3 +923,361 @@ def test_create_object_with_constrained_permission(self):
+            job_class="-",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            approval_required=True,
+            start_time=datetime.now(),
+        )
+        ScheduledJob.objects.create(
+            name="test4",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="-",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            approval_required=False,
+            start_time=datetime.now(),
+        )
+
+    def test_only_approvable_is_listed(self):
+        self.add_permissions("extras.view_scheduledjob")
+
+        response = self.client.get(self._get_url("list"))
+        self.assertHttpStatus(response, 200)
+        self.assertNotIn("test4", extract_page_body(response.content.decode(response.charset)))
+
+
+class JobResultTestCase(
+    TestJobMixin,
+    ViewTestCases.GetObjectViewTestCase,
+    ViewTestCases.ListObjectsViewTestCase,
+    ViewTestCases.DeleteObjectViewTestCase,
+    ViewTestCases.BulkDeleteObjectsViewTestCase,
+):
+    model = JobResult
+
+    @classmethod
+    def setUpTestData(cls):
+        obj_type = ContentType.objects.get(app_label="extras", model="job")
+        JobResult.objects.create(
+            name="local/test_view/TestJob",
+            job_id=uuid.uuid4(),
+            obj_type=obj_type,
+        )
+        JobResult.objects.create(
+            name="local/test_view/TestJob2",
+            job_id=uuid.uuid4(),
+            obj_type=obj_type,
+        )
+        JobResult.objects.create(
+            name="local/test_view/TestJob3",
+            job_id=uuid.uuid4(),
+            obj_type=obj_type,
+        )
+
+
+class JobTestCase(
+    TestCase,
+):
+    """
+    The Job view test cases.
+
+    Since Job is not an actual model, we have to improvise and test the views
+    manually.
+    """
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
+    def test_list_without_permission(self):
+        self.assertHttpStatus(self.client.get(reverse("extras:job_list")), 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_list(self):
+        response = self.client.get(reverse("extras:job_list"))
+        self.assertHttpStatus(response, 200)
+
+        response_body = extract_page_body(response.content.decode(response.charset))
+        self.assertIn("TestPass", response_body)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[], JOBS_ROOT=DUMMY_JOBS)
+    def test_get_without_permission(self):
+        response = self.client.get(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+        self.assertHttpStatus(response, 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], JOBS_ROOT=DUMMY_JOBS)
+    def test_get(self):
+        response = self.client.get(reverse("extras:job", kwargs={"class_path": "local/test_pass/TestPass"}))
+        self.assertHttpStatus(response, 200)
```

Comment:
Looks like this is failing in Travis - for some reason we're getting a 404 here instead of a 200.